rust/src/dns/detect.rs

```diff
@@ -89,6 +133,11 @@ fn match_opcode(detect: &DetectDnsOpcode, flags: u16) -> bool {
     }
 }
 
+fn match_rcode(detect: &DetectDnsRcode, flags: u16) -> bool {
+    let rcode = ((flags >> 11) & 0xf) as u16;
+    detect.rcode == rcode
```

Comment:
This is looking at the opcode, not the rcode. As the `rcode` is the last 4 bits we can drop the shift and just mask off the first 12 bits to get our rcode. See `pub fn rcode` in `dns.rs` for an example.

In fact, as `match_rcode` is called from `rs_dns_rcode_match` which already has the `tx`, you can just call `tx.rcode()` to get the rcode value instead of extracting it from the flags yourself.

Reply:
Is it 4 bites or a u16 cf https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-6 ?

Reply:
Oh seeing the comment Support for OPT RR from RFC6891 will be needed to parse RCODE values over 15

Reply:
would it be better if I used something like this instead:
`let rcode = (flags & 0x000f) as DetectUintData<u16>;`